### PR TITLE
Infer `config()` and `io()` names from assignment

### PR DESCRIPTION
**Pull request by akhilles**
> [!NOTE]
> **Medium Risk**
> Changes core Zen/Starlark parameter declaration semantics by introducing deferred resolution via `export_as`, which could affect module signatures and input resolution edge cases despite added tests and docs updates.
> 
> **Overview**
> **Zen parameter declarations now support name inference.** `config()` and `builtin.io()` accept an alternate calling form without an explicit `name` (e.g. `foo = config(str)` / `SIG = io(Net)`), resolving the parameter name from the top-level assignment.
> 
> Implementation refactors parameter handling into new `lang/param_decl.rs`, replacing the inlined `config()`/`declare_io()` logic with shared parsing + deferred `DeferredParameter` values that finalize on `export_as` and then record metadata/diagnostics.
> 
> Updates stdlib `input()`/`output()` wrappers, docs/spec, changelog, and adds tests covering inferred names, `checks` handling (including positional `None`), and ensuring unused nameless declarations are ignored. Also bumps the workspace `starlark-rust` git revision in `Cargo.toml`/`Cargo.lock`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8b2455a92583a7d8625144272aaf3681e3363d91. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---

<a href="https://app.devin.ai/review/diodeinc/pcb/pull/718" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,11 +11,11 @@ authors = ["Diode Computers, Inc. <founders@diode.computer>"]
...
-starlark = { git = "https://github.com/diodeinc/starlark-rust", rev = "41c8e1b317996a7b71f482886e66f89aa006272e" }
-starlark_map = { git = "https://github.com/diodeinc/starlark-rust", rev = "41c8e1b317996a7b71f482886e66f89aa006272e" }
-starlark_derive = { git = "https://github.com/diodeinc/starlark-rust", rev = "41c8e1b317996a7b71f482886e66f89aa006272e" }
-starlark_syntax = { git = "https://github.com/diodeinc/starlark-rust", rev = "41c8e1b317996a7b71f482886e66f89aa006272e" }
-allocative = { git = "https://github.com/diodeinc/starlark-rust", rev = "41c8e1b317996a7b71f482886e66f89aa006272e" }
+starlark = { git = "https://github.com/diodeinc/starlark-rust", rev = "be171dcffa01b85d8827e3fa1e2afe7c638ceabe" }
+starlark_map = { git = "https://github.com/diodeinc/starlark-rust", rev = "be171dcffa01b85d8827e3fa1e2afe7c638ceabe" }
+starlark_derive = { git = "https://github.com/diodeinc/starlark-rust", rev = "be171dcffa01b85d8827e3fa1e2afe7c638ceabe" }
+starlark_syntax = { git = "https://github.com/diodeinc/starlark-rust", rev = "be171dcffa01b85d8827e3fa1e2afe7c638ceabe" }
+allocative = { git = "https://github.com/diodeinc/starlark-rust", rev = "be171dcffa01b85d8827e3fa1e2afe7c638ceabe" }
```
Comment (LK) [inline Cargo.toml]: approved https://github.com/diodeinc/starlark-rust/pull/7#pullrequestreview-4101169957 if you want to merge that and move to main commit

**Review by LK — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (akhilles) [reply Cargo.toml]: done
